src/WebJobs.Script.WebHost/Security/KeyManagement/SimpleKubernetesClient.cs

```diff
@@ -0,0 +1,307 @@
+        private const string CaFile = "/run/secrets/kubernetes.io/serviceaccount/ca.crt";
+        private const string KubernetesSecretsDir = "/run/secrets/functions-keys";
+        private readonly HttpClient _httpClient;
+        private readonly IEnvironment _environment;
+        private Action _watchCallback;
+        private bool _disposed;
+        private AutoRecoveringFileSystemWatcher _fileWatcher;
+
+        public SimpleKubernetesClient(IEnvironment environment) : this(environment, CreateHttpClient())
+        { }
+
+        // for testing
+        internal SimpleKubernetesClient(IEnvironment environment, HttpClient client)
+        {
+            _httpClient = client;
+            _environment = environment;
+            Task.Run(() => RunWatcher());
+        }
+
+        private string KubernetesObjectName => _environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsKubernetesSecretName);
+
+        public bool IsWritable => !string.IsNullOrEmpty(KubernetesObjectName);
+
+        public async Task<IDictionary<string, string>> GetSecrets()
+        {
+            if (string.IsNullOrEmpty(KubernetesObjectName) && FileUtility.DirectoryExists(KubernetesSecretsDir))
+            {
+                return await GetFromFiles(KubernetesSecretsDir);
+            }
+            else if (!string.IsNullOrEmpty(KubernetesObjectName))
+            {
+                return await GetFromApiServer(KubernetesObjectName);
+            }
+            else
+            {
+                throw new InvalidOperationException($"{nameof(KubernetesSecretsRepository)} requires setting {EnvironmentSettingNames.AzureWebJobsKubernetesSecretName} or mounting secrets to {KubernetesSecretsDir}");
+            }
+        }
+
+        public async Task UpdateSecrets(IDictionary<string, string> data)
+        {
+            (var url, var isSecret) = await GetObjectUrl(KubernetesObjectName);
+            await CreateIfDoesntExist(url, isSecret);
+
+            data = isSecret
+                ? data.ToDictionary(k => k.Key, v => Convert.ToBase64String(Encoding.UTF8.GetBytes(v.Value)))
+                : data;
+
+            using (var request = await GetRequest(HttpMethod.Patch, url, new[] { new { op = "replace", path = "/data", value = data } }, "application/json-patch+json"))
+            {
+                var response = await _httpClient.SendAsync(request);
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        public void OnSecretChange(Action callback)
+        {
+            _watchCallback = callback;
+        }
+
+        private async Task RunWatcher()
+        {
+            if (string.IsNullOrEmpty(KubernetesObjectName) && FileUtility.DirectoryExists(KubernetesSecretsDir))
+            {
+                _fileWatcher = new AutoRecoveringFileSystemWatcher(KubernetesSecretsDir);
+                _fileWatcher.Changed += (object sender, FileSystemEventArgs e)
+                    => _watchCallback?.Invoke();
```

Comment:
@fabiocav This doesn't actually work. I can touch and change files in `KubernetesSecretsDir` and this never fires. I'm not sure what I'm doing wrong, or if the kernel used on my cluster has inotify disabled? I'm not sure

Reply:
I would assume something with the environment/OS. Does this work for you on local tests?

Reply:
I think you're right. It works locally on my machine. Though I think file change notifications can be turned off in linux. It's possible that either the file system or the kernel I'm running on on AKS doesn't fire the right notification.